app/controllers/ApplicationController.scala

```diff
@@ -69,10 +69,26 @@ trait ApplicationController extends BaseController with GatekeeperAuthWrapper {
       val applicationFuture = applicationService.fetchApplication(appId)
       val subscriptionsFuture = applicationService.fetchApplicationSubscriptions(appId)
 
+      def latestTOUAgreement(appWithHistory: ApplicationWithHistory): Option[TermsOfUseAgreement] = {
+        appWithHistory.application.checkInformation match {
+          case Some(checkInformation) => {
+            checkInformation.termsOfUseAgreements match {
+              case Nil => None
+              case agreements =>  Some(agreements.maxBy(_.timeStamp))
```

Comment:
This looks like flatmap 😃 

Plus, I think you want to use Option so, `Option(agreements.maxBy(_.timeStamp))` to handle the empty state. Some can be unpredictable!